### PR TITLE
Remove most code related to "classic" Oak

### PR DESCRIPTION
**Pull request by tiziano88**
Not used any more, and just taking up space and mental overhead, plus
additional CI time and compatibility burden when upgrading deps.

### CODE REVIEW

**Comment by tiziano88** [comment]
Note for reviewers:

I would like to get approval from all of you before merging.

I will remove the now-unused "required" checks from CI just before submitting.

There are a lot of files removed, and the GitHub UI struggles a bit; I suggest waiting a few seconds until all the diffs are loaded, whlie in files changed view, then look at the few files that got their contents changed, and then read through the names of the deleted files. If you think anything should not be deleted, please let me know, probably best if in a top-level comment as part of the review reply.

Thanks!

**Comment by ipetr0v** [comment]
Can we just move this into a separate repo like `project-oak/oak-classic` or `project-oak/oak-archive` ?
Because we might still use some of it at some point.

Just too sad to completely remove so much work.

**Review by jul-sh — APPROVED** [review]
LGTM. I would think of it less as "removing it forever" and more as in "setting it to rest in the version history, waiting for the right time to be resurrected"

**Comment by tiziano88** [comment]
> Did we already close the issues related to this code only?

Not yet, I think there are still a fair number of issues that we can close after merging this.

**Comment by tiziano88** [comment]
> Can we also simplify our [Dockerfile](https://github.com/project-oak/oak/blob/main/Dockerfile)?

Yes but I would prefer doing that in a separate PR.

**Review by mariaschett — APPROVED** [review]
Amazing! -84k LoC.

File: oak_functions/README.md
Comment (rbehjati) [inline oak_functions/README.md]: I think it would be useful to add some details from the deleted documentation here. For instance how to run an application or create configuration files. There is also some technical background that could be relevant, for instance about the use of WebAssembly. I'd suggest creating an issue to do this in a separate PR.

File: runner/src/internal.rs
Comment (rbehjati) [inline runner/src/internal.rs]: This files also has a server variant related to the introspection-client that caused some complication, particularly related to running tests for all features. That, and the code related to it, can be removed too.

**Review by rbehjati — APPROVED** [review]
Great. Thanks. I am very happy to see that prost in third_party is gone. We should celebrate this 🎉 

I think in the top-level README we have some links to some of the deleted documentation. Perhaps create an issue to fix the documentation after this is merged.

File: scripts/build_gh_pages
Comment (tiziano88) [reply scripts/build_gh_pages]: Fixed doc generation now.

File: scripts/common
Comment (tiziano88) [reply scripts/common]: Not any more.

File: oak_functions/README.md
Comment (tiziano88) [reply oak_functions/README.md:140]: Thanks, created https://github.com/project-oak/oak/issues/2541

**Pull request merged**
